test/chan.js

```diff
@@ -0,0 +1,69 @@
+var nano = require('..');
+var test = require('tape');
+
+test('adding and removing subscription channels', function (t) {
+
+  //t.plan(7);
+
+  var pub = nano.socket('pub');
+
+  //register some channels: hello, foo, bar
+  var sub = nano.socket('sub', {
+    chan: ['hello', 'foo', 'bar']
+  });
+
+  pub.bind('inproc://filter');
+  sub.connect('inproc://filter');
+
+  //the main object we'll use to establish channel registration and removal
+  var msgs = ['hello world','hello world','hello world','bar world','foo world'];
+  var msgqty = -1; //starting here so first msg counted can be zero
+  var sent = 0;
+
+  sub.on('message', function (buf) {
+
+    var msg = String(buf);
+
+    //increment msgqty when a msg is received (happening about every 100ms)
+    //we'll call finish() on the 5th msg received (case 4, see line 68)
+    switch(++msgqty){
+      case 0: return t.equal(msg, msgs[msgqty]); //msgs[0], 'hello world'
+      case 1: return t.equal(msg, msgs[msgqty]); //msgs[1], 'hello world'
+      case 2: return t.equal(msg, msgs[msgqty]); //msgs[2], 'hello world'
+      case 3: return t.equal(msg, msgs[msgqty]); //msgs[3], 'bar world'
+      case 4: return finish (msg);
+    }
+
+  });
+
+  setTimeout(send,        0);   //send msgs[0], 'hello world'
+  setTimeout(send,        100); //send msgs[1], 'hello world'
+  setTimeout(send,        200); //send msgs[2], 'hello world'
+  setTimeout(removeHello, 300); //send msgs[3], 'bar world' and remove hello
+  setTimeout(send,        400); //send msgs[4], 'foo world'
+
+  function send(){
+    pub.send(msgs[sent++]); //lazy incrementing
+  }
+
+  function removeHello(){
+
+    //stop listening for msg prefix: hello
+    sub.rmchan('hello');
+
+    //publish about 10 extra hello worlds to see if we can increment msgqty
+    var i = 0;
+    while(i++ < 10) pub.send('hello world');
+
+    send(); // send something with a registered prefix: 'bar world'
+  }
+
+  function finish(msg){
+    t.equal(msg, msgs[msgqty]); //'foo world'
+    t.equal(msg, msgs[4]); //msgqty received is only 4
+    t.equal(msg, 'foo world'); //prove case 4 is m4, the fifth msg
+    sub.close();
+    pub.close();
+    t.end()
+  }
+});
```

Comment:
ok but

> * show that constructing a socket without explicitly subscribing to any channel receives at least 2 messages of different prefixes.

seems like redundant coverage in `test/ipc.js`, `test/inproc.js`, `test/tcp.js`